shared/strings_util.h

```diff
@@ -5,6 +5,13 @@
 #include <vector>
 #include <sstream>
 
+inline bool string_endswith(std::string const &fullString, std::string const &ending) {
```

Comment:
while you're moving this who not rename it to string_ends_with

Reply:
Ok, I changed it to be conforming to the C++20 specification.
https://en.cppreference.com/w/cpp/string/basic_string/ends_with